concent_api/verifier/tasks.py

```diff
@@ -106,264 +107,252 @@ def blender_verification_order(
+
+    delete_source_files(package_paths_to_downloaded_archive_names[source_package_path])
+
+    blender_output_file_name = generate_full_blender_output_file_name(scene_file, frame_number, output_format.lower())
+    try_to_upload_blender_output_file(blender_output_file_name, output_format, subtask_id)
+
+    image_1, image_2 = load_images(
+        blender_output_file_name,
+        package_paths_to_downloaded_archive_names[result_package_path],
+        subtask_id
+    )
+    if not are_image_sizes_and_color_channels_equal(image_1, image_2):
+        log_string_message(
+            logger,
+            f'Blender verification failed. Sizes in pixels of images are not equal. SUBTASK_ID: {subtask_id}.'
+            f'VerificationResult: {VerificationResult.MISMATCH.name}'
         )
-        # If Blender finishes with errors, verification ends here
-        # Verification_result informing about the error is sent to the work queue.
-        if completed_process.returncode != 0:
-            log_string_message(
-                logger,
-                'Blender finished with errors',
-                f'SUBTASK_ID: {subtask_id}.'
-                f'Returncode: {str(completed_process.returncode)}.'
-                f'stderr: {str(completed_process.stderr)}.'
-                f'stdout: {str(completed_process.stdout)}.'
-            )
-            verification_result.delay(
-                subtask_id,
-                VerificationResult.ERROR.name,
-                str(completed_process.stderr),
-                ErrorCode.VERIFIER_RUNNING_BLENDER_FAILED.name
-            )
-            return
-    except SubprocessError as exception:
-        log_string_message(logger, f'Blender finished with errors. Error: {exception} SUBTASK_ID {subtask_id}')
         verification_result.delay(
             subtask_id,
-            VerificationResult.ERROR.name,
-            str(exception),
-            ErrorCode.VERIFIER_RUNNING_BLENDER_FAILED.name
+            VerificationResult.MISMATCH.name,
         )
         return
 
-    # Verifier deletes source files of the Blender project from its storage.
-    # At this point there must be source files in VERIFIER_STORAGE_PATH otherwise verification should fail before.
-    source_files_list = get_files_list_from_archive(
-        generate_verifier_storage_file_path(source_package_path)
-    )
-    for file_path in source_files_list + [package_paths_to_downloaded_file_names[source_package_path]]:
-        delete_file(file_path)
-    log_string_message(
-        logger,
-        f'Blender finished successfully. SUBTASK_ID: {subtask_id}.'
-        f'VerificationResult: {VerificationResult.MATCH.name}'
-    )
+    compare_images(image_1, image_2, subtask_id)
 
-    blender_output_file_name = generate_blender_output_file_name(scene_file)
-    upload_file_name = generate_upload_file_name(subtask_id, output_format)
 
-    # Read Blender output file.
+def compare_images(image_1, image_2, subtask_id):
```

Comment:
All of the functions below should be in utils.

Reply:
ok, I will move it and refactor tests (a loooooot of patches to change;))

Reply:
Right, but this can be done with few good `ctrl + R` ;) But I think `tasks.py` should contain only tasks functions.

Reply:
done